shared/viewmodels/RoomDirectoryViewModel.js

```diff
@@ -78,6 +64,23 @@ class RoomDirectoryViewModel extends ViewModel {
       })
     );
 
+    this._rooms = new ObservableArray(
+      rooms.map((room) => {
+        return {
+          roomId: room.room_id,
+          canonicalAlias: room.canonical_alias,
+          name: room.name,
+          mxcAvatarUrl: room.avatar_url,
+          homeserverUrlToPullMediaFrom: homeserverUrl,
+          numJoinedMembers: room.num_joined_members,
+          topic: room.topic,
+          archiveRoomUrl: matrixPublicArchiveURLCreator.archiveUrlForRoom(room.room_id, {
+            viaServers: [this.pageSearchParameters.homeserver],
```

Comment:
Adding the `?via` query parameter messes with the client-side [`ArchiveHistory`](https://github.com/matrix-org/matrix-public-archive/blob/d19c08397a9c4b9a9be091b714a8a664ba66b88f/shared/lib/archive-history.js) and causes the URL to be duplicated.

`http://localhost:3050/!xxx:server/date/2022/09/20?via=my.synapse.server` -> `http://localhost:3050/!xxx:server/date/2022/09/20?via=my.synapse.serverhttp://localhost:3050/!xxx:server/date/2022/07/29`

Reply:
Fixed in https://github.com/matrix-org/matrix-public-archive/pull/110